### PR TITLE
implement Share component.

### PR DESCRIPTION
**Pull request by hcnimi**

### CODE REVIEW

File: client/src/components/Share.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  render() {
+    return (
+      <div className="share">
+        <Helmet>
+          <meta property="og:url" content={'"' + this.props.file.url + '"'}/>
+          <meta property="og:type" content={'"' + this.props.file.type + '"'} />
+          <meta property="og:title" content={'"' + this.props.file.fileName + '"'} />
+        </Helmet>
+        <Col sm="auto">
+          <span class="fb-share-button" data-href="https://127.0.0.1/home/this.props.file.objectId" data-layout="button" data-size="small" data-mobile-iframe="false"><a target="_blank" href={'"https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2F127.0.0.1%2Fhome%2F' + this.props.file.objectId + '&amp;src=sdkpreparse" class="fb-xfbml-parse-ignore"'}>Share</a></span>
```
Comment (skoshkarli) [inline client/src/components/Share.jsx]: i am guessing this shouldn't be hardcoded to 127.0.0.1?

**Comment by skoshkarli** [comment]
config has already been added. shows a conflict now

**Comment by hcnimi** [comment]
Resolved conflicts

File: client/src/components/Share.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import Helmet from 'react-helmet';
+import { Row, Col, Progress } from 'reactstrap';
+import css from '../styles/Share.css';
+
+class Share extends React.Component {
```
[the remark this answers is not in the document]
Comment (hcnimi) [reply client/src/components/Share.jsx]: Added handleClick.

File: .gitignore [hunk 3]
```diff
@@ -64,5 +64,6 @@ typings/
 client/dist/*
 !client/dist/index.html
 
+
```
Comment (skoshkarli) [inline .gitignore]: non-blocking: remove extra unnecessary space.

File: server/index.js [hunk 4]
```diff
@@ -140,10 +140,6 @@ app.post('/upload', upload.single('file'), function(req, res, next) {
   res.send('Successfully uploaded ' + req.file.length + ' files!')
 });
 
-app.get('*', (req, res) => {
```
Comment (skoshkarli) [inline server/index.js]: why do you want to remove this? Wont' it break the app for all non-supported routes?

File: server/index.js [hunk 5]
```diff
@@ -140,7 +140,7 @@ app.post('/upload', upload.single('file'), function(req, res, next) {
...
+app.get('/*', (req, res) => {
```
Comment (skoshkarli) [inline server/index.js]: not sure about this, just wondering what your thoughts are:
* vs /* 
will * catch all routes including the / ?
will */ catch all routes including the / ?

File: client/src/components/Share.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (skoshkarli) [reply client/src/components/Share.jsx]: 👍

File: server/index.js (hunk 5 above)
Comment (jbgomez) [reply server/index.js]: If it's a catch-all, I think we should just go with `*`

File: client/src/components/Share.jsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      method: 'share',
+      display: 'popup',
+      href: 'https://developers.facebook.com/docs/',
+    }, function(response){});
+  }
+  render () {
+    return (
+      <div className="share">
+        <Helmet>
+          <meta property="og:url" content={'"' + this.props.file.url + '"'}/>
```
Comment (skoshkarli) [inline client/src/components/Share.jsx]: question: how are we generating the url which we are sharing and where is it stored to get the file later?

**Pull request closed without merging**